tratto/src/main/java/star/tratto/util/javaparser/TypeUtils.java

```diff
@@ -0,0 +1,295 @@
+     * returns an empty string.
+     *
+     * @param classGetName a ClassGetName form of a type
+     * @return the package name of the class
+     */
+    public static String getPackageNameFromClassGetName(String classGetName) {
+        List<String> nameSegments = Arrays.asList(classGetName.split("\\."));
+        List<String> packageSegments = nameSegments.subList(0, nameSegments.size() - 1);
+        return String.join(".", packageSegments);
+    }
+
+    /**
+     * Gets the innermost class from a ClassGetName form of a type.
+     *
+     * @param classGetName a ClassGetName form of a type
+     * @return the innermost class of the type
+     */
+    public static String getInnermostClassNameFromClassGetName(String classGetName) {
+        List<String> nameSegments = Arrays.asList(classGetName.split("[.$]"));
+        return nameSegments.get(nameSegments.size() - 1);
+    }
+
+    /**
+     * Returns the array level of the type.
+     *
+     * @param typeName a ClassGetName or ClassGetSimpleName form of a type
+     * @return the array level of the type
+     */
+    private static int getArrayLevel(String typeName) {
+        int arrayLevel = 0;
+        for (int i = 0; i < typeName.length(); i++) {
+            if (typeName.charAt(i) == '[') {
+                arrayLevel++;
+            }
+        }
+        return arrayLevel;
+    }
+
+    /**
+     * Adds pairs of square brackets to a given ClassGetSimpleName type.
+     *
+     * @param classGetSimpleName a ClassGetSimpleName form of a type
+     * @param arrayLevel number of array levels to add
+     * @return the simple type name with the added number of array levels
+     */
+    private static String addArrayLevel(String classGetSimpleName, int arrayLevel) {
+        return classGetSimpleName + "[]".repeat(arrayLevel);
+    }
+
+    /**
+     * Returns true iff the ClassGetName component type is a primitive type.
+     * The ClassGetName form of a type uses field descriptors for arrays of
+     * primitive types.
+     *
+     * @param classGetNameComponent a ClassGetName type
+     * @return true iff the ClassGetName component is a field descriptor
+     * @see TypeUtils#primitiveFieldDescriptors
+     */
+    private static boolean isPrimitiveFieldDescriptor(String classGetNameComponent) {
+        return primitiveFieldDescriptors.contains(classGetNameComponent);
+    }
+
+    /**
+     * Gets the ClassGetName form of the base component type for a given
+     * ClassGetName array type. For example,
+     *     "[[I"    =>    "I"
+     *     "[Ljava.lang.String;"    =>    "java.lang.String"
+     *
+     * @param classGetNameArray a ClassGetName form of an array type
+     * @return the ClassGetName form of the component type without arrays
+     */
+    private static String classGetNameComponentType(String classGetNameArray) {
+        int arrayLevel = getArrayLevel(classGetNameArray);
+        if (arrayLevel == 0) {
+            return classGetNameArray;
```

Comment:
For "Ljava.lang.String;", this returns "Ljava.lang.String;" whereas I think it should return ""java.lang.String".